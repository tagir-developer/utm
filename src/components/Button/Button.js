import React from 'react'

const Button = ({ text, type, handler}) => {
	return (

		<button
			type="button"
			className={`btn btn-${type} mr-3 my-2`}
			onClick={handler}
		>
			{text}
		</button>

	)
}

Button.defaultProps = {
	text: 'Текст кнопки',
	type: 'success',
	handler: () => {}
}

export default Button