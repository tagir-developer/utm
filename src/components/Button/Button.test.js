import React from 'react'
import Button from './Button'


describe('<Button />', () => {

	let wrapper

	beforeEach(() => {
		wrapper = shallow(<Button />)
	})

	describe('Component render. Component props testing', () => {

		it('Should Button component render without props', () => {
			expect(toJson(wrapper)).toMatchSnapshot()
		})


		it('Should render Button type', () => {
			wrapper.setProps({
				type: 'danger'
			})
			expect(wrapper.find('button').props().className).toContain(' btn-danger ')
		})

		it('Should render Button text', () => {
			wrapper.setProps({
				text: 'Тестовый текст кнопки'
			})
			expect(wrapper.find('button').text()).toBe('Тестовый текст кнопки')
		})

	})

	describe('Button component handlers testing', () => {

		it('Should call "ButtonClose" handler', () => {
			const clickHandler = jest.fn()
			wrapper.setProps({
				handler: clickHandler
			})
			wrapper.find('button').simulate('click')
			expect(clickHandler).toHaveBeenCalled()
		})

		it('Should use default handler', () => {
			const result = Button.defaultProps.handler()
			expect(result).toBe(undefined)
		})

	})


})