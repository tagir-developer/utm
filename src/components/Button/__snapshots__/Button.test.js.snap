// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Component render. Component props testing Should Button component render without props 1`] = `
<button
  className="btn btn-success mr-3 my-2"
  onClick={[Function]}
  type="button"
>
  Текст кнопки
</button>
`;
